src/nupic/regions/SPRegion.cpp

```diff
@@ -997,4 +997,33 @@ void SPRegion::deserialize(BundleIO &bundle) {
     sp_ = nullptr;
 }
 
+bool SPRegion::operator==(const RegionImpl &o) const {
```

Comment:
thanks for this function! Please have a look at one of the SP tests, I think there's an implementation of this method (can be compared, and removed from the test then)

Reply:
>  I think there's an implementation of this method 

There might have been at some point.  I looked for it and could not find it.
Perhaps you are thinking of the compares in SpatialPoolerTest.  Those are not quite the same.

Reply:
> thinking of the compares in SpatialPoolerTest. Those are not quite the same.

yes, I meant those. Sorry